src/libutil/util.cc

```diff
@@ -1034,6 +1034,19 @@ std::vector<char *> stringsToCharPtrs(const Strings & ss)
     return res;
 }
 
+// return true if the programmed returned ok, false otherwise
+bool getProgramStatus(Path program, bool searchPath, const Strings & args,
```

Comment:
I didn't find a function that allow to check the output of a program, but I will be suprised if there aren't any in this code base.